src/ExpressionEvaluator/CSharp/Source/ExpressionCompiler/EvaluationContext.cs

```diff
@@ -81,13 +78,20 @@ internal sealed class EvaluationContext : EvaluationContextBase
                 previous.Compilation :
                 metadataBlocks.ToCompilation();
```

Comment:
Consider:  Would it be simpler to keep just one overload of CreateTypeContext, remove the previous parameter and pull this check out into the caller?